### PR TITLE
[Merged by Bors] - chore: move LinearOrder Nat instance to Mathlib.Data.Nat.Defs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

Preliminary to removing the order dependency from `Mathlib/Data/Nat/Defs.lean`, but that needs a little more work

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/UnionFind.lean [hunk 1]
```diff
@@ -217,7 +217,7 @@ def findAux (self : UnionFind α) (x : Fin self.size) :
       m.Models self.arr ∧ m'.Models s ∧ m'.rank = m.rank ∧
       (∃ hr, (m'.parent ⟨root, hr⟩).1 = root) ∧
       m.rank x ≤ m.rank root := by
-  let y := self.arr[x].parent
+  let y := (self.arr[x]'(Fin.isLt x)).parent
```
Comment (YaelDillies) [inline Mathlib/Data/UnionFind.lean:220]: What's this notation even?

File: Mathlib/Order/Nat.lean [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+-/
+
+import Mathlib.Init.Data.Nat.Notation
+import Mathlib.Init.Order.Defs
+
+/-!
+# LinearOrder for ℕ
+
+This file shows that ℕ is a linear order.
+-/
```
Comment (YaelDillies) [inline Mathlib/Order/Nat.lean:14]: Can you mimic the docstring of `Algebra.Order.Group.Nat`?

File: Mathlib/Data/UnionFind.lean
Comment (YaelDillies) [inline Mathlib/Data/UnionFind.lean:1]: Given that you haven't managed to get rid of the order imports here, maybe you should keep `Order.Nat` imported?

Comment (Ruben-VandeVelde) [reply Mathlib/Data/UnionFind.lean:1]: It turns out I was really close, but I'd be fine reverting it all as well.

File: Mathlib/Data/UnionFind.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Data/UnionFind.lean:223]: Array indexing with explicit proof that the index is in range. I didn't check why this worked before.

File: Mathlib/Order/Lattice.lean [hunk 3]
```diff
@@ -897,7 +897,6 @@ instance (priority := 100) {α : Type u} [LinearOrder α] : DistribLattice α wh
     | Or.inl h => inf_le_of_left_le <| sup_le_sup_left (le_inf (le_refl b) h) _
     | Or.inr h => inf_le_of_right_le <| sup_le_sup_left (le_inf h (le_refl c)) _
 
-instance : DistribLattice ℕ := inferInstance
```
Comment (YaelDillies) [inline Mathlib/Order/Lattice.lean:900]: Where is that gone?

File: Mathlib/Order/Nat.lean [hunk 4]
```diff
@@ -29,4 +29,256 @@ instance instOrderBot : OrderBot ℕ where
...
+  le_antisymm := @Nat.le_antisymm
+  le_total := @Nat.le_total
+  lt := Nat.lt
+  lt_iff_le_not_le := @Nat.lt_iff_le_not_le
+  decidableLT := inferInstance
+  decidableLE := inferInstance
+  decidableEq := inferInstance
+#align nat.linear_order Nat.instLinearOrder
+
+instance : DistribLattice ℕ := inferInstance
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Nat.lean:45]: Over here.

**Comment by Ruben-VandeVelde** [comment]
This was starting to get too big, so I reverted to a more minimal change: the LinearOrder instance temporarily goes to `Data.Nat.Defs` and I'll move it to `Order.Nat` in #13092

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Pull request closed without merging**